packages/drivers/odsp-driver/src/epochTracker.ts

```diff
@@ -143,9 +144,11 @@ export class EpochTracker implements IPersistedFileCache {
         const request = this.addEpochInRequest(url, fetchOptions, addInBody);
         let epochFromResponse: string | undefined;
         try {
-            const response = await this.rateLimiter.schedule(
+            const response = await handleReadOnlyServerState(async () => this.rateLimiter.schedule(
```

Comment:
I'm not sure I like it being here. Can it be incorporated into getWithRetryForTokenRefresh?
Is the concern that we only want to apply it to few calls, not to all SPO calls? It feels like it should be fine for it to apply to all calls that can fall into this bucket.

Reply:
The other alternative - RetryCoherencyErrorsStorageAdapter that I've added to handle 409 errors. But it feels like it might be well suited for this case too (might want to rename class, or follow similar pattern and add another adapter on top of it - I'd prefer single class if code composes well).

Reply:
I added this in epochTracker because that is the choke point for all the storage calls even those made outside of storage service like ops and create new. This will enable this feature for all calls.

Reply:
I don't think "getWithRetryForTokenRefresh" is right place because that is for token refresh.

Reply:
Yes, RetryCoherencyErrorsStorageAdapter will not capture file creation (as an example)

Is it for all calls, or only calls that write (modify) file?
Glancing through usage of getWithRetryForTokenRefresh, do we have problems in any of these flows?
- I assume it should not apply to flows like getFileLinkCore and similar, right?
- What about connectToDeltaStream - can joinSession or PUSH connection return with same error? Should we handle it there too?
- readTree -> fetchSnapshot

Reply:
Yes, join session can have this issue as I have been told. But not the fetch snapshot and getblobs apis as those are purely read apis.